build.ps1

```diff
@@ -1,33 +1,177 @@
 ## Copyright (c) Microsoft Corporation.
 ## Licensed under the MIT License.
-
-[CmdletBinding()]
+# default behavior is build only
+[CmdletBinding(SupportsShouldProcess=$true)]
 param (
     [Parameter()]
     [string]
-    $Configuration = "Debug",
+    $Configuration = "Release",
+
+    [Parameter()]
+    [switch]$Publish,
+
+    [Parameter()]
+    [switch]$Test,
+
+    [Parameter()]
+    [switch]$NoBuild,
 
     [Parameter()]
-    [switch]
-    $Clean
+    [switch]$Clean
 )
 
-try {
-    Push-Location "$PSScriptRoot/src/code"
+$moduleName = "Microsoft.PowerShell.TextUtility"
+$projectRoot = $PSScriptRoot
+
+$srcRoot = Join-Path $ProjectRoot src
+$codeRoot = Join-Path $srcRoot code
+$tstRoot = Join-Path $ProjectRoot test
+$modManifest = Join-Path $srcRoot "${ModuleName}.psd1"
+
+
+function AssertFile {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]$Path
+    )
+
+    if (-not (Test-Path $Path)) {
```

Comment:
Is this function needed instead of just using `Test-Path -ErrorAction Stop`?

Reply:
`test-path -erroraction stop` does not throw